cmake/FindMKL.cmake

```
@@ -84,7 +85,7 @@ if (ENABLE_MKLCPU_BACKEND OR ENABLE_MKLGPU_BACKEND)
     list(APPEND MKL_LINK_C ${TBB_LINK})
   endif()
   if(ENABLE_MKLGPU_BACKEND)
-    set(MKL_LINK_SYCL ${MKL_LINK_PREFIX} ${LIB_PREFIX}${MKL_SYCL} ${MKL_LINK_C} ${LIB_PREFIX}${OPENCL_LIBNAME})
+    set(MKL_LINK_SYCL ${MKL_LINK_PREFIX} ${LIB_PREFIX}${MKL_SYCL} ${MKL_LINK_C} ${LIB_PREFIX}${OPENCL_LIBNAME} ${SYCL_LINK_FLAGS})
```

Don't we use DPC++ compiler here? It should have -fsycl by default shouldn't it?
But even if we have to provide explicitly linker option for some reasons I think we need to change it a little bit. I'm not sure I like the idea of providing linker option as one of library because in some cases order of options and libraries is important. I'd suggest at least add -fsycl to the very beginning of the MKL_LINK_SYCL before any libraries.
```suggestion
    set(MKL_LINK_SYCL ${SYCL_LINK_FLAGS} ${MKL_LINK_PREFIX} ${LIB_PREFIX}${MKL_SYCL} ${MKL_LINK_C} ${LIB_PREFIX}${OPENCL_LIBNAME})
```

---

Thanks for the feedback. We are using clang++ compiler, so it does not have -fsycl by default. For this file, compilation has -fsycl but not link line. 
I will update the sequence with your suggestion.

I will change the order.